src/nvim/ex_cmds.c

```diff
@@ -2116,6 +2116,16 @@ do_ecmd (
     }
   }
 
+  // Make re-editing a terminal buffer a no-op
+  if (!other_file && curbuf->terminal != NULL) {
+      // this is needed for when we are called by do_argfile() and the new
+      // argument index becomes the terminal buffer we are already editing
+      check_arg_idx(curwin);
+      maketitle();
```

Comment:
Normally the index in the argument list isn't switched when the buffer is not reloaded.